modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -417,6 +417,23 @@ const openSelectionModal = ({
 
 						const allSelectedNodes = allSelectedElements.getDOMNodes();
 
+						const searchContainerId = searchContainer.get('id');
+
+						if (
+							!Liferay.SPA &&
+							sessionStorage.getItem(
+								searchContainerId +
+									Liferay.ThemeDisplay.getUserId() +
+									'_selections'
+							)
+						) {
+							sessionStorage.removeItem(
+								searchContainerId +
+									Liferay.ThemeDisplay.getUserId() +
+									'_selections'
+							);
+						}
+
```

Comment:
We clean up, because once the modal's select method runs, the user is "done" with its selections, the user initiated an action to run on the selections he/she made. If we don't clean up here, the selections would remain in sessionStorage and would be selected again if the user opens the same selectionModal.

Reply:
Is that a bad thing? :thinking: If it is, is it a bad thing that we are not cleaning up on any other navigation?

Maybe what we want is to limit session storage behavior only on pagination.